packages/wonder-blocks-layout/components/media-layout.test.js

```diff
@@ -0,0 +1,343 @@
+                        },
+                    }),
+                    medium: StyleSheet.create({
+                        test: {
+                            color: "teal",
+                        },
+                    }),
+                    large: StyleSheet.create({
+                        test: {
+                            color: "magenta",
+                        },
+                    }),
+                };
+                resizeWindow(size);
+
+                // Act
+                const style = getStyle(styleSheets);
+                const color = {
+                    medium: "teal",
+                    large: "magenta",
+                }[size];
+
+                // Assert
+                expect(style.color).toEqual(color);
+            });
+        }
+    });
+
+    describe("window resizing", () => {
+        it("should update the style when the window gets smaller", async () => {
+            // Arrange
+            const styleSheets = {
+                large: StyleSheet.create({
+                    test: {
+                        color: "blue",
+                    },
+                }),
+                small: StyleSheet.create({
+                    test: {
+                        color: "orange",
+                    },
+                }),
+            };
+
+            resizeWindow("large");
+
+            const wrapper = mount(
+                <MediaLayout styleSheets={styleSheets}>
+                    {({mediaSize, mediaSpec, styles}) => {
+                        return <View style={styles.test}>Hello, world!</View>;
+                    }}
+                </MediaLayout>,
+            );
+
+            resizeWindow("small");
+            wrapper.update();
+
+            // Act
+            const style = wrapper.find("div").prop("style");
+
+            // Assert
+            expect(style.color).toBe("orange");
+        });
+
+        it("should update the style when the window gets larger", async () => {
+            // Arrange
+            const styleSheets = {
+                large: StyleSheet.create({
+                    test: {
+                        color: "blue",
+                    },
+                }),
+                small: StyleSheet.create({
+                    test: {
+                        color: "orange",
+                    },
+                }),
+            };
+
+            resizeWindow("small");
+
+            const wrapper = mount(
+                <MediaLayout styleSheets={styleSheets}>
+                    {({mediaSize, mediaSpec, styles}) => {
+                        return <View style={styles.test}>Hello, world!</View>;
+                    }}
+                </MediaLayout>,
+            );
+
+            resizeWindow("large");
+            wrapper.update();
+
+            // Act
+            const style = wrapper.find("div").prop("style");
+
+            // Assert
+            expect(style.color).toBe("blue");
+        });
+    });
+});
```

Comment:
Love the tests!

I think some of them have their "act" code in the "arrange" section and could do with some tweaks.